tls/extensions/s2n_certificate_extensions.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "error/s2n_errno.h"
+#include "utils/s2n_safety.h"
+#include "stuffer/s2n_stuffer.h"
+
+#include "tls/s2n_tls.h"
+#include "tls/s2n_tls13.h"
+#include "tls/extensions/s2n_server_status_request.h"
+#include "tls/extensions/s2n_server_sct_list.h"
+
+static int s2n_get_number_certs_in_chain(struct s2n_cert *head, uint8_t *chain_length);
+
+int s2n_certificate_extensions_parse(struct s2n_connection *conn, struct s2n_blob *extensions)
```

Comment:
perhaps this should be `s2n_server_certificate_extensions_parse` to be more specific

Reply:
both server and client parse extensions

Reply:
We can keep this function although I believe both server and client would handle extensions a little differently eg. I'm not sure if a client would want to parse a "Signed Certificate Timestamp" extension from the server. Since `s2n_certificate_extensions_parse()` is called by `s2n_x509_validator_validate_cert_chain()` that could operate in both server/client modes may require us to check for conn->mode instead.